src/scripts/update-firmware-binaries.js

```diff
@@ -56,101 +57,109 @@ const DEFAULT_SETTINGS = {
 		}
 	},
 };
+/* eslint-enable */
 
 let versionTag = '';
-const githubAPI = new github();
+const githubAPI = new Github();
 
-const exitWithMessage = (message: string): void => {
-	console.log(message);
-	process.exit(0);
+const exitWithMessage = (message: string) => {
+  console.log(message);
+  process.exit(0);
 };
 
-const exitWithJSON = (json: Object): void => {
-	exitWithMessage(JSON.stringify(json, null, 2));
-}
-
-const downloadFile = (url: string): Promise<*> => {
-	return new Promise((resolve, reject) => {
-		const filename = nullthrows(url.match(/.*\/(.*)/))[1];
-		console.log('Downloading ' + filename + '...');
-		const file = fs.createWriteStream(
-      settings.BINARIES_DIRECTORY +
-      '/' +
-      filename
-    );
-    file.on('finish', () => file.close(() => resolve(filename)));
-		request(url).pipe(file).on('error', exitWithJSON);
-	});
+const exitWithJSON = (json: Object) => {
+  exitWithMessage(JSON.stringify(json, null, 2));
 };
 
+const downloadFile = (url: string): Promise<*> =>
+  new Promise((resolve: (filename: string) => void) => {
+    const filename = nullthrows(url.match(/.*\/(.*)/))[1];
+    console.log(`Downloading ${filename}...`);
+
+    const file = fs.createWriteStream(
+      `${settings.BINARIES_DIRECTORY}/${filename}`,
+    );
+
+    file.on('finish', (): void => file.close((): void => resolve(filename)));
+    request(url).pipe(file).on('error', exitWithJSON);
+  });
+
+
 const downloadFirmwareBinaries = async (
   assets: Array<Asset>,
-): Promise<*> => {
-	const assetFileNames = await Promise.all(assets.map(asset => {
-		if (asset.name.match(/^system-part/)) {
-			return downloadFile(asset.browser_download_url);
-		}
-    return '';
-	}));
+): Promise<Array<string>> => {
+  const assetFileNames = await Promise.all(
+    assets.map((asset: Object): Promise<?string> => {
+      if (asset.name.match(/^system-part/)) {
+        return downloadFile(asset.browser_download_url);
+      }
+      return Promise.resolve('');
+    }),
+  );
 
-  return assetFileNames.filter(item => item);
+  return assetFileNames.filter((item: ?string): boolean => !!item);
 };
 
 const updateSettings = (): Array<string> => {
-	let versionNumber = versionTag;
-	if (versionNumber[0] === 'v') {
-		versionNumber = versionNumber.substr(1);
-	}
+  let versionNumber = versionTag;
+  if (versionNumber[0] === 'v') {
+    versionNumber = versionNumber.substr(1);
+  }
 
-	let settings = JSON.stringify(
+  let scriptSettings = JSON.stringify(
```

Comment:
not sure how this should be named. it was `settings` and it was shadowing with imported app settings.